File: evals/fixtures/controlled/frontend/src/app/models/incident.ts
export type IncidentSeverity = "informational" | "warning" | "critical";
export type IncidentStatus = "open" | "acknowledged" | "resolved";

export interface Incident {
  id: string;
  service: string;
  severity: IncidentSeverity;
  status: IncidentStatus;
  createdAt: string;
}

export const requiresImmediateEscalation = (incident: Incident): boolean =>
  incident.severity === "critical" && incident.status === "open";

File: evals/fixtures/controlled/backend/src/SignalDesk.Domain/Incidents/Incident.cs
namespace SignalDesk.Domain.Incidents;

public sealed record Incident(
    Guid Id,
    string Service,
    IncidentSeverity Severity,
    IncidentStatus Status,
    DateTimeOffset CreatedAt)
{
    public bool RequiresImmediateEscalation =>
        Severity == IncidentSeverity.Critical && Status == IncidentStatus.Open;
}
